authors/apps/articles/migrations/0001_initial.py

```diff
@@ -1,6 +1,5 @@
-# Generated by Django 2.1.7 on 2019-04-05 07:52
```

Comment:
Please explain the reason for the changes in this migration file and the steps you went through to introduce those changes

Reply:
I changed the migrations because at first, our Article model used the User instance as the foreign key and after getting the new updates of profile I had to change the user instance to Profile instance. But since the current migrations also were using the User model I had to reset them so that they can use Profile instance also. Without resetting them I would end up with errors for every operation I carried out since the migrations where expecting User instance and I was feeding in Profile instance. 
Then I used this link on how to reset them.
https://simpleisbetterthancomplex.com/tutorial/2016/07/26/how-to-reset-migrations.html.

Reply:
Thanks for the info.
However, when working with migrations it is better to generate a new file rather than making adjustments to an old one. This will ensure that we do not tamper with existing data in the database among other reasons.

Therefore, please remove any changes you've made to this file then run `python manage.py makemigrations` to generate a new migration file that you can push as part of your implementation.

Reply:
Thanks @RonKbS, at first I followed that implementation but I could result in errors. And I found the same issue raised here https://code.djangoproject.com/ticket/22319

Reply:
Kindly elaborate on the errors that were being generated following that implementation.

Reply:
It's the ValueError that happens on the line below whereby migrations ask for an instance that was changed to Profile instance.
```dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]
```

Reply:
I just ran migrations successfully on your branch using that particular method and was successful (I reverted the changes in this migration file you that you had implemented).
<img width="1403" alt="Screenshot 2019-04-15 at 17 42 23" src="https://user-images.githubusercontent.com/40039818/56141786-29159180-5fa6-11e9-84e2-5b16bc6de4e1.png">

Can you list the exact steps you ran that resulted in the error being raised?

Reply:
@RonKbS Kindly run the tests with that new implementation because I too don't get any error in making and migrating migrations. But after running tests, this is the error that arises.
![error](https://user-images.githubusercontent.com/22131999/56143352-0f297e00-5fa9-11e9-994e-a405c1ea3f6c.PNG)